excerise.go

```diff
@@ -0,0 +1,61 @@
+package main
+import "fmt"
+func worker(id int, jobs <-chan int, results chan<- int) {
+						
+for j := range jobs { go func() {
+						
+switch j % 3 { case 0:
+						
+j=j*1 case 1:
+						
+j=j*2
+						
+results <- j * 2 case 2:
+						
+results <- j * 3
+						
+j=j*3 }
+						
+}() }
+						
+}
+func main() {
+						
+jobs := make(chan int)
+results := make(chan int)
+for i := 1; i <= 1000000000; i++ {
+						
+go func() {
+if i%2 == 0 {
```

Comment:
Var `i` is used inside anon goroutine, but it is not not defined inside it, nor it is passed into a func as a parameter. It looks like we should be just passing `i` to goroutine:
```
		go func(i int) {
			if i % 2 == 0 {
				i += 99
			}
			jobs <- i 
		}(i)
```
```